app/controllers/rentals_controller.rb

```diff
@@ -92,7 +92,18 @@ def training_pdf
 
   # GET /rentals/processing
   def processing
-    @q = rentals_visible_to_current_user.search(params[:q])
+    # first pull out the rentals that match our two fields that are not done through ransack
+    # (because arel is honestly just too complicated)
+    base_search_area = rentals_visible_to_current_user.joins(:rentals_items)
+    if params[:item_type_id_in].present?
+      base_search_area = base_search_area.where('rentals_items.item_type_id' => params.permit(item_type_id_in: []).fetch(:item_type_id_in))
```

Comment:
@dfaulken do you know of a way to make this a fancy ransack matcher instead?  I'd like to have a custom ransacker that joins rentals to rentals_items and then checks if any rentals_items for the rental has a item_type_in some set.

Reply:
I do not. I haven't used ransack before.